### PR TITLE
Guard the variance square root against exactly-zero variance

### PR DESCRIPTION
**Pull request by Tianbuyu-wwx**
Guard the variance square root against exactly-zero variance.

Sites
1. `src/timesfm/torch/util.py`, `update_running_stats` (2 lines): `inc_var` and `new_var` are clamped
   only at `0.0`, and `0.0` is exactly where `sqrt` has an infinite derivative.
2. `v1/src/timesfm/pytorch_patched_decoder.py` and `v1/src/timesfm/patched_decoder.py` (1 line each):
   `masked_var` is likewise clamped at `0.0` before `sqrt` / `jnp.sqrt`.

Why it matters
A patch whose values are all identical has an exactly-zero variance. The forward pass returns `0.0` and
stays finite, but the backward pass returns `inf` → `NaN`, so a training loop that guards on the loss
silently discards the step. Constant patches are not exotic in real inputs: censored monitoring data
reported at the detection limit, intermittent demand (runs of zeros), night-time zero output.

This is the same class of failure that #341 addressed for the variance formula and the sigma clamp; the
square roots themselves were not covered there. The codebase already uses a small positive floor for
sigma (`torch.clamp(sigma, min=self.config.tolerance)`, from #341) — this PR applies the same idea at the
variance roots.

Reproduce
```python
import torch
from timesfm.torch import util
x = torch.full((4, 2, 32), 0.5, requires_grad=True)     # one distinct value
n, mu, sigma = (torch.zeros(4, 2) for _ in range(3))
_, (_, _, out) = util.update_running_stats(n, mu, torch.ones(4, 2), x, torch.zeros_like(x, dtype=torch.bool))
out.sum().backward()
print(x.grad)                    # [nan, nan, nan] before this patch
```

Test
`src/timesfm/torch/util_test.py` (3 tests, CPU and CUDA; `0.5` is used because its patch sum is exactly
representable, so the variance is exactly zero on both backends). Verified in a local clone at `8cb0628`:

* before the patch: `FAILED (failures=3)`
* after the patch: `Ran 3 tests in 0.307s — OK`

No downstream number changes: for any `sigma ≥ 1e-4` the added `1e-8` is below float32 resolution
relative to `sigma`; only the case `sigma == 0` moves from `0.0` (NaN derivative) to `1e-4` (finite
derivative).

### CODE REVIEW

**Comment by Tianbuyu-wwx** [comment]
The `Python package build` workflow is showing `action_required` with no jobs started, so the test suite hasn't actually run on this PR yet. Could a maintainer approve the workflow run?

Also, #404 touches the same `update_running_stats` function — happy to rebase if it lands first. And if a named constant is preferred over the `1e-8` literal, I'll adjust.
